.github/workflows/build-and-deploy.yml

```
@@ -0,0 +1,102 @@
+
+      - name: Check for new commits before building on schedule
+        if: github.event_name == 'schedule' && steps.last_successful_commit.outputs.last_commit_sha == steps.latest_commit.outputs.latest_commit_sha
+        uses: andymckay/cancel-action@0.2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+        with:
+          install: true
+
+      - name: Log in to Docker Hub
+        uses: docker/login-action@v2
```

actionlint (1.7.7)

the runner of "docker/login-action@v2" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)